lohica-js-network/src/components/RegForm.js

```diff
@@ -1,162 +1,123 @@
+import GenderRadio from './inputComponents/GenderRadio';
+import AgeInput from './inputComponents/AgeInput';
+import ImageInput from './inputComponents/ImageInput';
+ 
 
 class RegistrationForm extends  React.Component {
 
-    constructor(props, context){
+    constructor(props){
         super(props);
-        this.state = {
-            name: '',
-            surname: '',
-            midName: '',
-            email: '',
-            age: '' 
-        }
-
-
-        // bind this to the class method
-        this.handleChange = this.handleChange.bind(this);
-
-        this.formFieldValidation = this.formFieldValidation.bind(this);
 
+        this.handleSubmit = this.handleSubmit.bind(this);
     }
 
-    formFieldValidation (field) {
-        console.log(this.state);
-        let charNum = this.state[field].length;
-        if(charNum > 10) {
-            return 'success';
-        } else if(charNum > 5) {
-            return 'warning';
-        } else if(charNum > 0) {
-            return 'error';
+    handleSubmit(e){
+        let inputFieldsData = this.props.store.getState().formInput;
+        e.preventDefault();
+        if(inputFieldsData.emailValid && inputFieldsData.imageValid && inputFieldsData.nameValid && inputFieldsData.surnameValid && inputFieldsData.midNameValid) {
+            // user info
+            let userName = inputFieldsData.name;
+            let userSurname = inputFieldsData.surname;
+            let userMidName = inputFieldsData.midName;
+            let userEmail = inputFieldsData.email;
+            let userGender = inputFieldsData.gender;
+            let userAge = inputFieldsData.age;
+
+            let data = new FormData();
+
+            data.append('file', this.props.inputState.imageData);
+            data.append('name', userName);
+            data.append('surname', userSurname);
+            data.append('midname', userMidName);
+            data.append('email', userEmail);
+            data.append('gender', userGender);
+            data.append('age', userAge);
+
+            fetch('/signup',{
+                method: 'POST',
+                body: data
+            }).then((mes) => mes.json())
+                .then(data => console.log(data))
+                .catch(err => console.log(err));
         }
-        return null;
     }
 
-    // set state is async - to see value - use callback or async await
-    handleChange(e) {
-        let key = e.target.name;
-        let val = e.target.value;
-        this.setState({[key]: val});
-    }
-
-    render() {
+    render() {  
+        // console.log(this.props);
 
         // age options
         let Options = [];
         for(let i =1; i<=99; i++){
             Options.push(`${i}`);
-        }   
-
-        let label = 'label';
+        }
         return (
         <Form horizontal>
             <Grid>
                 <Row>
 
-                    <RegistrationInput 
-                        size = "small"
+                    <NameInput
```

Comment:
Since you use NameInput for the bunch of different inputs - i'd recommend you to rename NameInput to something like TextInput

Reply:
revoved nameinput component - did not have enough control over props - npw have seperate components for each input field (properly named)